### ISSUE TITLE
Improvements to Rule Calculation Order for Intermediaries

### ISSUE THREAD

**Issue by jsexauer** [scikit-fuzzy/scikit-fuzzy#76]
The current way the control module figures out the order to process rules is based on if the terms that are in the predicate of the rule have been fully calculated at least once.  This is not a good implementation because:
 - It produces inconsistent results when multiple rules drive the same variable's term.  Eg:
```python
rule1 = fuzz.Rule(service['good'], tip['good'])
rule2 = fuzz.Rule(quality['good'], tip['good'])
```
- It produces inconsistent results when multiple rules drive the same variable and the membership functions of the term overlaps.  Eg:
```python
tip['good'] = fuzz.trimf(tip.universe, [13, 25, 25])
tip['fair'] = fuzz.trimf(tip.universe, [10, 20, 25]) # Exaggerated overlap in membership
rule1 = fuzz.Rule(service['good'], tip['good'])
rule2 = fuzz.Rule(quality['good'], tip['fair'])
```

Note that these limitations only affect intermediary variables.

A better approach is to ensure all rules which drive the input side of an intermediary (regardless of term) are calculated before any rules which use the intermediary in the predicate.

**Comment by JDWarner** [comment]
Closing as I believe this was addressed in #79. Feel free to reopen if this remains an issue.

**Comment by nilayrox18** [comment]
sir i am getting what you are saying but my issue is arise when i pass my
input to a system..

tipping.input['quality'] = 6.5
it take s quality as a index and show error like 'ValuesView' does not
support indexing....

please fix this sir..


On Fri, Mar 11, 2016 at 11:30 PM, Josh Warner <notifications@github.com>
wrote:

> Closing as I believe this was addressed in #79
> <https://github.com/scikit-fuzzy/scikit-fuzzy/pull/79>. Feel free to
> reopen if this remains an issue.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/scikit-fuzzy/scikit-fuzzy/issues/76#issuecomment-195479623>
> .
>



-- 
*Mr. Nilay S. Shah*
*M.Tech ICT  *
*Contact no : +91-8866319714*
